### ISSUE TITLE
Method to check if repo exists without cloning.

### ISSUE THREAD

**Issue by talk2cerlin** [czproject/git-php#7]
i wanted a way to check if the repo exists in the git url

Use case:
1. Clones base project (kinda big project)
2. Setting up the base project
3. Clones child repo

I dont want to come to third step and fail if repo is not found and do a clean up. Child repo has multiple implementations and the name of child repo is given by the user (chances of typo)

I was thinking if there is any way to verify that the repo is existing and my cli app can read its info (which proves me that the user has valid ssh key).

Basically i wanted to start cloning/downloading the repos after verifying that all the repos are existing and accessible.

I came across [this](http://stackoverflow.com/questions/23914896/check-that-git-repository-exists) question.

**Comment by janpecha** [comment]
Interesting issue. What about [this](https://github.com/czproject/git-php/commit/7475f7fa459e66398eecd6ee412cd31dd262cb3e#diff-86989dba81f1844b0566b8e2f1285dfcR638) (commit in branch `wip-is-remote-readable`)? It's quick&dirty solution but I haven't better idea.  If you know a better solution, please send a PR.

**Comment by talk2cerlin** [comment]
Looks good. Will check and update you soon

**Comment by lplume** [comment]
parameters should be added as well

**Comment by janpecha** [comment]
Good point, I will update it.

**Comment by lplume** [comment]
got few time to test it today, im not sure that return code === 0 means the url is readable. Maybe it is, but consider calling git ls-remote along with a refs that does not exist? The command does not exit with error, the url is "readable" but not clonable.

Thanks for the effort, i like the way you wrap git.

**Comment by janpecha** [comment]
You are right, in this case is repository "readable" but not "clonable". I updated it and added new parameter 
`$refs`.

I merged it to `master` branch and released version `3.9.0-RC1`.

Please try it, if it's ok now. Thanks.

**Comment by lplume** [comment]
Thanks a lot, just in time, gonna try in few hours, will give you feedbacks
